test/reporters/tests/default.test.ts

```diff
@@ -238,6 +239,21 @@ describe('default reporter', async () => {
     expect(stdout).toContain('Example project')
     expect(stdout).toContain('\x1B[30m\x1B[45m Example project \x1B[49m\x1B[39m')
   })
+
+  test('extended reporter can override getFullName', async () => {
+    class Custom extends DefaultReporter {
+      getFullName(test: RunnerTask, separator?: string): string {
+        return `${separator}{ name: ${test.name}, meta: ${test.meta.custom} } (Custom getFullName here)`
+      }
+    }
+
+    const { stderr } = await runVitest({
+      root: 'fixtures/metadata',
+      reporters: new Custom(),
+    })
+
+    expect(stderr).toMatch('FAIL   > { name: fails, meta: Failing test added this } (Custom getFullName here')
```

Comment:
@lafeuil from https://github.com/vitest-dev/vitest/issues/8018 👋 

Is this the kind of use case you are trying to achieve?

Reply:
Yes, it is exactly that :+1: